### ISSUE TITLE
[Feature] Change video file extension, in desiredCapabilities

### ISSUE THREAD

**Issue by victorMerlis** [zalando/zalenium#202]
elgalu/docker-selenium supporting this feature -> https://github.com/elgalu/docker-selenium#record-videos

**Comment by elgalu** [comment]
Hi @victorMerlis , right now we actually save it in mkv to limit the amount of CPU usage and then we convert it to mp4 afterwards (after the tests finished)

Which video file extension do you want to have in the end? instead of the current mp4 target? and why? because just changing the extension doesn't necessarily mean it will change the encoding.

**Comment by victorMerlis** [comment]
I want some light format, this is the order of the video formats based on size.
Ogg < 3gp < Flv < Xvid < Rm < Vob < Wmv < Mp4 < Mpeg < Avi < Mkv
lets say Flv?
this will reduce compression time and most important file size

**Comment by elgalu** [comment]
Another reason we chose the mp4 as the final format was that the html5 video player worked fine with it, at Zalenium dashboard.html

**Comment by victorMerlis** [comment]
Good point :)
Right now (as far as I know) there isn't any option to save the videos only for the failed tests, I saw your suggestion to run the tests without recording and to enable it if one of the tests failed. It's not practical if you automate your tests :\
So if I automate my tests it's more important (at least to me) to be able to save the videos no matter if it's failed or pass, each video should be max 1M (and not +20M)

**Comment by elgalu** [comment]
Let's wait for @diemol to be back from holidays and see what he thinks.

What would also help here is some bench-marking to compare `Ogg < 3gp < Flv < Xvid < Rm < Vob < Wmv < Mp4 < Mpeg < Avi < Mkv` CPU usage during test time, and produced video file size. All for a relevant test. There should be major visual changes, if we save a test whose pages don't change much they compression will win but will be miss real world tests examples IMHO)

**Comment by elgalu** [comment]
And we should also put in that feature matrix which formats are supported for html5 video player, that work at least on Chrome which is the main browser we expect Zalenium users to use while opening the dashboard.html

**Comment by diemol** [comment]
I think the solution for this would be the implementation of #151, which is something I want to do but for several reasons during these weeks (vacation, marriage, etc...) I have not found the time.

We will start looking seriously the backlog of issues again in September to start implementing more helpful features (#151 is also needed internally, so we will definitively implement it).

Would this solution suit you @victorMerlis?

**Comment by victorMerlis** [comment]
Sorry for the late response, I was on vacation.
@diemol congratulation for your marriage :)
#151 is a good feature but not a solution, because if there will be let's say 10 failed tests and each test video is about 20M...

**Comment by diemol** [comment]
Thanks for the greetings @victorMerlis!

I am not sure how to proceed here, I understand that it is nice to have small file sizes but for now we don't have the resources to investigate and come up with a smaller file size (and perhaps extension) that would work naturally in the dashboard.

That would also need some post processing task as well, I guess.

Let's leave this open for now and reassess in a while to see if we have a better idea.

**Comment by TomLarrow** [comment]
Just tossing out an idea, This container: https://hub.docker.com/r/gadjet/video_transcoding/ which uses this project https://github.com/donmelton/video_transcoding might be a possibility. We've recently disabled video recording as it was putting too big of a strain on our system when we had a dozen tests finish up relatively close time frame and all want to re-encode their video simultaneously. 

If things were just saved as large files which required relatively low CPU to generate, and then a second process came along in a single threaded manner and recompressed these down that might save an entire cluster from being crushed as they all tried to recompress in parallel.  Sure the video wouldn't be available right away, but they could eventually appear in a smaller format. Perhaps the 'convert-video' function of the application which can convert videos from Matroska to MP4 format without transcoding the video might make the larger files instantly available before the recompression process can shrink them.

**Comment by diemol** [comment]
Closing, as it is consolidated in #645
If development starts, we'll reopen the issue.

**Comment by tbhatt-conga** [comment]
@diemol - is there any update for having the compressed size of video to handle 1000-2000 testcases.

**Comment by diemol** [comment]
https://github.com/zalando/zalenium/issues/654#issuecomment-632814276
